### PR TITLE
Add basic OTP support

### PR DESCRIPTION
**Pull request by robin-nitrokey**
Fixes https://github.com/Nitrokey/nitrokey-app2/issues/77

### CODE REVIEW

**Comment by robin-nitrokey** [comment]
Updated.  Limitations:
- only TOTP
- no PIN cache
- no connection cache
- very basic UI

**Comment by robin-nitrokey** [comment]
Added HOTP support.  It’s not that slow even though there is no connection cache so I think this could be merged now.

**Comment by robin-nitrokey** [comment]
Updated:
- bump pynitrokey to 0.4.38
- enforce minimum app version
- use list properties to replace the second list call for protected credentials

What’s your opinion on the PR?  Can we merge it soon so that we can base other changes on it?  From my POV, the feature is pretty stable now.

File: nitrokeyapp/secrets_tab/__init__.py [hunk 1]
```diff
@@ -0,0 +1,277 @@
...
+        visible = credential is not None and credential.otp is not None
+        self.ui.pushButtonOtpGenerate.setVisible(visible)
+
+    @pyqtSlot()
+    def update_otp_timeout(self) -> None:
+        if not self.otp_timeout:
+            return
+        timeout = (self.otp_timeout - datetime.now()).total_seconds()
+        if timeout >= 0:
+            self.ui.progressBarOtpTimeout.setValue(timeout)
```
Comment (Niklap97) [inline nitrokeyapp/secrets_tab/__init__.py:234]: is the progressbar working for you? 
I get a `TypeError: setValue(self, value: int): argument 1 has unexpected type 'float'`
So maybe just make it `self.ui.progressBarOtpTimeout.setValue(int(timeout))` ?

Comment (robin-nitrokey) [reply nitrokeyapp/secrets_tab/__init__.py:234]: Works for me, maybe we have different PyQt versions.  But it should not hurt so I’ll change it anyway.

**Review by daringer — APPROVED** [review]

**Pull request merged**